### PR TITLE
util/dbterror: migrate test-infra to testify

### LINKED ISSUE #26174: migrate test-infra to testify for `util/dbterror` pkg

### PR DESCRIPTION
**Pull request by Fedomn**
### What problem does this PR solve?

Issue Number: close #26174 

### What's Changed:

Migrate test-infra to testify for util/dbterror pkg.

### Release note 

- No release note

### CODE REVIEW

File: util/dbterror/terror_test.go [hunk 1]
```diff
@@ -47,44 +41,44 @@ func (s *testkSuite) TestErrorRedact(c *C) {
 	QuestionMark := "?"
 
 	err := class.NewStd(errno.ErrDupEntry).GenWithStackByArgs(SensitiveData, NoSensitiveValue)
-	c.Assert(strings.Contains(err.Error(), genErrMsg(errno.MySQLErrName[errno.ErrDupEntry].Raw, QuestionMark, NoSensitiveValue)), IsTrue)
+	assert.True(t, strings.Contains(err.Error(), genErrMsg(errno.MySQLErrName[errno.ErrDupEntry].Raw, QuestionMark, NoSensitiveValue)))
```
Comment (tisonkun) [inline util/dbterror/terror_test.go:44]: You can make use of `require.Contains` for a better assertion and fail report.

**Review by tisonkun — CHANGES_REQUESTED** [review]
Thanks for your contribution @Fedomn ! Generally looks good. One comment inline and `require` will call `FailNow` if assertion fails, which is the same manner as `c.Assert`.

File: util/dbterror/terror_test.go (hunk 1 above)
Comment (Fedomn) [reply util/dbterror/terror_test.go]: okay, thanks a lot, already updated

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by disksing — APPROVED** [review]

**Comment by Fedomn** [comment]
/c @tiancaiamao Could you help on reviewing this PR?

**Review by xhebox — APPROVED** [review]

**Pull request merged**